clicommand/bootstrap.go

```diff
@@ -148,6 +150,12 @@ var BootstrapCommand = cli.Command{
 			Usage:  "The name of the agent running the job",
 			EnvVar: "BUILDKITE_AGENT_NAME",
 		},
+		cli.StringFlag{
+			Name:   "queue",
+			Value:  "",
+			Usage:  "The name of the queue the agent belongs to, if tagged",
+			EnvVar: "BUILDKITE_AGENT_META_DATA_QUEUE",
```

Comment:
FYI to other watchers that this is configured through meta data [passed to the agent at runtime](https://buildkite.com/docs/pipelines/environment-variables#bk-env-vars-buildkite-agent-meta-data-star). GitHub seems to have gobbled my previous comment, but the gist is that this is typically set on the agent using `BUILDKITE_AGENT_TAGS`. It might be more reliable to pull from that value instead as it doesn't rely on a running job. It's worth being aware that you can configure multiple queues, I don't remember exactly how it would look but I suspect it's a comma separated value. Given that; it might also be interesting to try and record a value to capture what the job agent queue query was, but it would require a change on the Buildkite API side that I can organise.